### PR TITLE
Closes #392 | Create dataset loader for Generated Review ENTH

### LINKED ISSUE #392: Create dataset loader for generated_reviews_enth
Dataloader name: `generated_review_enth/generated_review_enth.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?generated_review_enth

|     Dataset| generated_review_enth |
|-------------|---|
| Description | This dataset (referred to as generated_reviews_yn in scb-mt-en-th-2020) are English product reviews generated by CTRL, translated by Google Translate API and annotated as accepted or rejected (correct) based on fluency and adequacy of the translation by human annotators. This allows it to be used for English-to-Thai translation quality esitmation (binary label), machine translation, and sentiment analysis. For SEACrowd, we use data with correct = 1. |
| Subsets | - |
| Languages | tha, eng |
| Tasks | Machine Translation |
| License | Creative Commons Attribution Share Alike 4.0 (cc-by-sa-4.0) |
| Homepage | https://github.com/vistec-ai/generated_reviews_enth |
| HF URL | - |
| Paper URL | https://arxiv.org/pdf/2007.03541.pdf |

### PR DESCRIPTION
**Pull request by khelli07**
Closes [#392](https://github.com/SEACrowd/seacrowd-datahub/issues/392)

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

image

### CODE REVIEW

File: seacrowd/sea_datasets/generated_review_enth/generated_review_enth.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+class GeneratedReviewENTHDataset(datasets.GeneratorBasedBuilder):
+    """This dataset (referred to as generated_reviews_yn in scb-mt-en-th-2020) are English product reviews generated by CTRL, translated by Google Translate API and annotated as accepted or rejected (correct) based on fluency and adequacy of the translation by human annotators. This allows it to be used for English-to-Thai translation quality esitmation (binary label), machine translation, and sentiment analysis."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name="generated_review_enth_source",
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/generated_review_enth/generated_review_enth.py:53]: I suggest using the `_DATASETNAME` here (and in all other instances)

File: seacrowd/sea_datasets/generated_review_enth/generated_review_enth.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "valid.jsonl"),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/generated_review_enth/generated_review_enth.py:125]: If you're not using the `split` keyword argument, I suggest removing it (and also in the `SplitGenerator`). The one on the template is just a guide on how it's used.

**Review by ljvmiranda921 — COMMENTED** [review]
Hi @khelli07 , thanks for the PR! I tested the data loader and it works properly. I suggest running the formatter `make check_file=...` since I encountered a few formatting errors when I ran it locally:

```
All done! ✨ 🍰 ✨
1 file reformatted.
isort seacrowd/sea_datasets/generated_review_enth/generated_review_enth.py
flake8 seacrowd/sea_datasets/generated_review_enth/generated_review_enth.py --max-line-length 250
seacrowd/sea_datasets/generated_review_enth/generated_review_enth.py:24:251: E501 line too long (546 > 250 characters)
seacrowd/sea_datasets/generated_review_enth/generated_review_enth.py:44:251: E501 line too long (419 > 250 characters)
make: *** [quality] Error 1
```

I also have a few nits in the source (see comments below)

**Review by ljvmiranda921 — APPROVED** [review]
LGTM ! Let's just wait for @tellarin 's review :)

**Review by elyanah-aco — COMMENTED** [review]
Hi @khelli07, thanks for the PR! Dataloader works well on my end.

Just a suggestion, since this dataset is part of scb-mt-en-th-2020, can you change `CITATION` to the one used [here](https://github.com/SEACrowd/seacrowd-datahub/blob/master/seacrowd/sea_datasets/scb_mt_en_th/scb_mt_en_th.py) instead?

**Comment by holylovenia** [comment]
A friendly reminder for @khelli07 to address @elyanah-aco's suggestion.

**Comment by holylovenia** [comment]
Just a heads up, @khelli07 will continue updating by the end of this week, @ljvmiranda921 @elyanah-aco.

**Review by elyanah-aco — APPROVED** [review]
Thanks for the edit @khelli07! Merging PR now

**Pull request merged**